src/uwtools/set_config.py

```diff
@@ -122,5 +128,25 @@ def create_config_obj(argv):
             out_object = config_obj
         out_object.dump_file(user_args.outfile)
 
+    ##begin show_format
+    # if --show_format, print a list of required configuration settings to stdout
+
+    if user_args.show_format:
+        # first ensure all required args are present
+        assert(
+            user_args.show_format and user_args.outfile
+        ), "args: --show_format also requires -outfile for reference"
+
+        if outfile_type != infile_type:
+            # next, provide needed format if not performing a simple conversion
+            convert_list = [".yaml", ".yml", ".bash", ".sh", ".ini", ".IN", ".nml"]
+            if outfile_type not in convert_list:
+                print(str(out_object.__doc__))
+            else:
+                print(f"Performing direct conversion from {infile_type} to {outfile_type}, \
+                    no additional formatting is required!")
+        else:
+            print("If not converting, no additional formatting is required!")
```

Comment:
I would simplify this logic even more and check to see whether the out_object has a __doc__ option. If so, log it (since logger has been implemented at this point). If it doesn't have the __doc__ option then, log something like "no additional information is provided for this output file type"

Reply:
Hmm, I had put this in originally essentially to return an answer rather than error if a user checked the config on say .yaml to .ini. I like checking for doc, but we have doc statements on all of the Config classes. I've tried a few tricks using decorators or subheadings in docstrings, but its not very good.

Reply:
Sorted this out and figured it out, now will print if the file_write docs are longer than default